server/src/helpers/getFriendsRecommendations.ts

```diff
@@ -0,0 +1,70 @@
+import { IUser } from '../db.requests/getFriendsRecommendations.requests';
+
+interface IGraph {
+    [id: string]: string[];
+}
+
+interface IUsersFriendshipDegree {
+    [id: string]: number;
+}
+
+interface IFriendsRecommendations {
+    [id: string]: number;
+}
+
+export const createGraph = (user: IUser): IGraph => {
+    const graph: IGraph = {
+        [user._id]: user.following.map((fol: any) => fol._id),
+    };
+
+    return user.following.reduce((result: IGraph, fol: any) => {
+        const updatedByFollowingResult = fol.following.reduce((res: IGraph, follow: any) => {
+            if (!res[follow]) {
+                return {...res, [follow]: [] };
+            }
+            return res;
+        }, result);
+
+        return {...updatedByFollowingResult, [fol._id]: fol.following.map((f: object) => f.toString())};
+    }, graph);
+};
+
+export const findRecommendations = (users: IGraph, root: string): IFriendsRecommendations => {
+    const usersFriendshipDegree: IUsersFriendshipDegree = {};
+    const friendsRecommendations: IFriendsRecommendations = {};
+
+    const allUsers = Object.keys(users);
+    allUsers.forEach((user: string) => usersFriendshipDegree[user] = Infinity);
```

Comment:
why Infinity ?

Reply:
Add comment to code. Infinity here means that node is have not been visited yet.